flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionTest.java

```diff
@@ -327,6 +329,48 @@ private void testReleaseMemory(final ResultPartitionType resultPartitionType) th
 		}
 	}
 
+	@Test
+	public void testPipelinedPartitionBufferPool() throws Exception {
+		testPartitionBufferPool(ResultPartitionType.PIPELINED_BOUNDED);
+	}
+
+	@Test
+	public void testBlockingPartitionBufferPool() throws Exception {
+		testPartitionBufferPool(ResultPartitionType.BLOCKING);
+	}
+
+	private void testPartitionBufferPool(ResultPartitionType type) throws Exception {
+		//setup
+		final int networkBuffersPerChannel = 2;
+		final int floatingNetworkBuffersPerGate = 8;
+		final NetworkBufferPool globalPool = new NetworkBufferPool(20, 1, 1);
+		final ResultPartition partition = new ResultPartitionBuilder()
+			.setResultPartitionType(type)
+			.setFileChannelManager(fileChannelManager)
+			.setNetworkBuffersPerChannel(networkBuffersPerChannel)
+			.setFloatingNetworkBuffersPerGate(floatingNetworkBuffersPerGate)
+			.setNetworkBufferPool(globalPool)
+			.build();
+
+		try {
+			partition.setup();
+			BufferPool bufferPool = partition.getBufferPool();
+			// verify the amount of buffers in created local pool
+			assertEquals(partition.getNumberOfSubpartitions() + 1, bufferPool.getNumberOfRequiredMemorySegments());
+			if (type.isBounded()) {
+				final int maxNumBuffers = networkBuffersPerChannel * partition.getNumberOfSubpartitions() + floatingNetworkBuffersPerGate;
+				assertEquals(maxNumBuffers, bufferPool.getMaxNumberOfMemorySegments());
+			} else {
+				assertEquals(Integer.MAX_VALUE, bufferPool.getMaxNumberOfMemorySegments());
+			}
+
+		} finally {
+			// cleanup
+			globalPool.destroyAllBufferPools();
```

Comment:
Again would be cool if we could use JUnit's setup and teardown mechanism instead of implementing it manually.

Reply:
Let me try that.

Reply:
After double checking, in this `ResultPartitionTest`, we have the `fileChannelManager` field for class-level, not for test-level. 

Furthermore only my added new tests would use `globalPool` actually,  and there were already many old tests which would not reuse this part now. So it might not be very suitable now to make it per-test `setup/teardown`.  We can do that if it is necessary to have more tests to reuse it future.